.buildkite/bk.integration.pipeline.yml

```diff
@@ -70,6 +73,9 @@ steps:
           - packaging-windows
         env:
           TEST_PACKAGE: "github.com/elastic/elastic-agent/testing/integration/leak"
+          CUSTOM_IMAGE_TAG: "git-${BUILDKITE_COMMIT:0:12}"
+          CI_ELASTIC_AGENT_DOCKER_IMAGE: "docker.elastic.co/beats-ci/elastic-agent-cloud"
```

Comment:
Why is CI_ELASTIC_AGENT_DOCKER_IMAGE in so many steps? IIUC this is only used by the mage load/push (and we could get rid of those as well in favor of docker load/tag/push)

I am pretty sure we don't package and push after provisioning ESS. Is this because of retries ? Even if it is we don't have to re-upload the same image, right ?